patches/server/0986-FOV-Scale-Implementation.patch

```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Moulberry <james.jenour@protonmail.com>
+Date: Tue, 6 Jun 2023 02:08:29 +0800
+Subject: [PATCH] FOV Scale Implementation
+
+Modifies the player's fov by a factor
+The multiplier is affected by the client Accessibility option 'FOV Effects'
+
+diff --git a/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java b/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java
+index a20bcf5877106ae00c83cf376a0dd16016cd1c7c..f262347fa9de6f3f9d118a1ffa810f7292558509 100644
+--- a/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java
++++ b/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java
+@@ -14,7 +14,7 @@ public class ClientboundPlayerAbilitiesPacket implements Packet<ClientGamePacket
+     private final boolean canFly;
+     private final boolean instabuild;
+     private final float flyingSpeed;
+-    private final float walkingSpeed;
++    public float walkingSpeed; // Paper - expose for ServerPlayer#onUpdateAbilities
```

Comment:
I don't know with mojang moving packets to records more and more, I honestly think your later suggestion would fit better than an AT. Cloning the abilities struct, or honestly, even just mutating the Abilities instance before constructing the packet and mutating it back to its original state after the packet construction would probably be a tad more future proof when they inevitably touch that packet and make it a record (all its fields are final already so it would be rather easy for them)